pkg/controller/kubelet-config/kubelet_config_nodes.go

```diff
@@ -0,0 +1,273 @@
+			return fmt.Errorf("unable to modify the kubelet configuration on the node, node condition status not ready")
+		}
+
+		var originalKubeConfig = new(v1beta1.KubeletConfiguration)
+
+		// updating the kubelet configuration with the Node specific configuration.
+		err = updateOriginalKubeConfigwithNodeConfig(node, originalKubeConfig)
+		if err != nil {
+			return err
+		}
+
+		byteData, err := json.Marshal(originalKubeConfig)
+		if err != nil {
+			return err
+		}
+		kc.Spec.KubeletConfig.Raw = byteData
+		kc.ObjectMeta.Annotations = map[string]string{
+			ctrlcommon.GeneratedByControllerVersionAnnotationKey: version.Hash,
+		}
+		// Create or Update, on conflict retry
+		if err := retry.RetryOnConflict(updateBackoff, func() error {
+			var err error
+			if isNotFound {
+				_, err = ctrl.client.MachineconfigurationV1().KubeletConfigs().Create(context.TODO(), kc, metav1.CreateOptions{})
+			} else {
+				_, err = ctrl.client.MachineconfigurationV1().KubeletConfigs().Update(context.TODO(), kc, metav1.UpdateOptions{})
+			}
+			return err
+		}); err != nil {
+			return fmt.Errorf("Could not Create/Update KubeletConfig: %v", err)
+		}
+		glog.Infof("Applied Node configuration %v on MachineConfigPool %v", key, pool.Name)
+
+		// updating the node status based on the latest kubelet config CR status
+		err = ctrl.updateNodestatus(kc, node)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (ctrl *Controller) enqueueNode(node *osev1.Node) {
+	key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(node)
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("Couldn't get key for object %#v: %v", node, err))
+		return
+	}
+	ctrl.nodeQueue.Add(key)
+}
+
+func (ctrl *Controller) updateNode(old, cur interface{}) {
+	oldNode := old.(*osev1.Node)
+	newNode := cur.(*osev1.Node)
+	if !reflect.DeepEqual(oldNode.Spec, newNode.Spec) {
+		// skipping the update in case of the Worker-Latency-Profile type transition from "Default" to "LowUpdateSlowReaction" and vice-versa
+		// (TODO) Ideally the user request has to be honoured, the transition need to be from Default -> Medium -> Low or vice-versa.
+		// Restricting the request for now until this process is automated in future.
+		if (oldNode.Spec.WorkerLatencyProfile == osev1.DefaultUpdateDefaultReaction && newNode.Spec.WorkerLatencyProfile == osev1.LowUpdateSlowReaction) || (oldNode.Spec.WorkerLatencyProfile == osev1.LowUpdateSlowReaction && newNode.Spec.WorkerLatencyProfile == osev1.DefaultUpdateDefaultReaction) {
+			glog.Infof("Skipping the Update Node event, name: %s, transition not allowed from old WorkerLatencyProfile: %s to new WorkerLatencyProfile: %s", newNode.Name, oldNode.Spec.WorkerLatencyProfile, newNode.Spec.WorkerLatencyProfile)
+			return
+		}
+		glog.V(4).Infof("Update Node event, name: %s", newNode.Name)
+		ctrl.enqueueNode(newNode)
+	}
+}
+
+func (ctrl *Controller) addNode(obj interface{}) {
+	node := obj.(*osev1.Node)
+	glog.V(4).Infof("Add Node event %s", node.Name)
```

Comment:
Same "Add Node event : node name" isn't really clear for debugging

Reply:
Addressed

Reply:
nice change :+1: